godtools/App/Share/Common/StringSearcher/StringSearcher.swift

```diff
@@ -24,6 +24,19 @@ class StringSearcher {
         }
     }
     
+    func search(for searchText: String, in searchables: [StringSearchable]) -> [StringSearchable] {
```

Comment:
I'm curious how to best resolve StringSearcher now.  It would be best to have a single method rather than the 2 methods.

I'm wondering was it a bad design and suggestion by me to have an interface for ToolFilterCategoryDomainModel?

On StringSearcher I lean more towards having method ```func search<T: StringSearchable>(for searchText: String, in searchables: [T]) -> [T]``` rather than ```func search(for searchText: String, in searchables: [StringSearchable]) -> [StringSearchable]``` because of the generics.

Reply:
Sorry @rachaelblue, I think it was a bad idea to suggest an interface. 

Did some searching on when to use inheritance vs an interface and found a helpful article.  To me ```Any``` looks like a specialized version of the filter option.   What are your thoughts on that?

<img width="835" alt="Screenshot 2024-07-26 at 11 21 54 AM" src="https://github.com/user-attachments/assets/e449a557-57ad-4a12-aa6a-78d6dc0d63dd">